client/tx/factory.go

```diff
@@ -281,6 +283,12 @@ func (f Factory) PreprocessTx(keyname string, builder client.TxBuilder) error {
 	return f.preprocessTxHook(f.chainID, key.GetType(), builder)
 }
 
+// WithExtensionOptions returns a Factory with given extension options added to the existing options.
```

Comment:
It would be interesting to explain here the use case for it.

Reply:
Added, see if it's ok.